contracts/swap-order-lock-script/src/main.rs

```diff
@@ -0,0 +1,112 @@
+//! Generated by capsule
+//!
+//! `main.rs` is used to define rust lang items and modules.
+//! See `entry.rs` for the `main` function.
+//! See `error.rs` for the `Error` type.
+
+#![no_std]
+#![no_main]
+#![feature(lang_items)]
+#![feature(alloc_error_handler)]
+#![feature(panic_info_message)]
+
+mod error;
+
+use core::result::Result;
+
+use num_bigint::BigUint;
+use share::cell::{OrderKind, SwapOrderLockArgs};
+use share::ckb_std::{
+    ckb_constants::Source,
+    ckb_types::prelude::*,
+    default_alloc,
+    high_level::{load_cell, load_cell_data, load_script, load_witness_args, QueryIter},
+};
+use share::{ckb_std, decode_u128, get_cell_type_hash};
+
+use crate::error::Error;
+
+// Alloc 4K fast HEAP + 2M HEAP to receives PrefilledData
+default_alloc!(4 * 1024, 2048 * 1024, 64);
+
+ckb_std::entry!(program_entry);
+
+/// program entry
+fn program_entry() -> i8 {
+    // Call main function and return error code
+    match main() {
+        Ok(_) => 0,
+        Err(err) => err as i8,
+    }
+}
+
+fn main() -> Result<(), Error> {
+    let script = load_script()?;
+    let lock_args = SwapOrderLockArgs::from_raw(script.args().as_slice())?;
+
+    for (idx, cell) in QueryIter::new(load_cell, Source::Input).enumerate() {
+        if cell.lock().code_hash().as_slice() == lock_args.user_lock_hash.as_ref() {
+            let witness = load_witness_args(idx, Source::Input)?;
+            if witness.total_size() != 0 {
+                return Ok(());
+            }
+        }
+    }
+
+    let mut index = 0;
+    let self_cell = load_cell(0, Source::GroupInput)?;
```

Comment:
If there are multiple cells in the same GroupInput, which means user place multiple same orders.
We should verify all of them in a loop instead of just first one.

My bad, i will update design doc to reflect this.